sherlockml/clients/report.py

```diff
@@ -0,0 +1,174 @@
+    author_id = fields.UUID(required=True)
+    report_path = fields.String(required=True)
+    report_key = fields.String(required=True)
+    report_bucket = fields.String(required=True)
+    notebook_path = fields.String(required=True)
+    report_id = fields.UUID()
+
+    @pre_load
+    def set_report_id(self, data):
+        if data.get("report_id") is None and "report_id" in self.context:
+            data["report_id"] = self.context["report_id"]
+        return data
+
+    @post_load
+    def make_report_version(self, data):
+        return ReportVersion(**data)
+
+
+class ReportSchema(Schema):
+    created_at = fields.DateTime(required=True)
+    name = fields.String(required=True, data_key="report_name")
+    id = fields.UUID(required=True, data_key="report_id")
+    description = fields.String(required=True)
+    active_version = fields.Nested(ReportVersionSchema, required=True)
+
+    @pre_load
+    def add_report_id_to_context(self, data):
+        if "report_id" in data:
+            self.context["report_id"] = data["report_id"]
+        return data
+
+    @post_load
+    def make_active_report(self, data):
+        return Report(**data)
+
+
+class VersionedReportSchema(Schema):
+
+    created_at = fields.DateTime(required=True)
+    name = fields.String(required=True, data_key="report_name")
+    id = fields.UUID(required=True, data_key="report_id")
+    description = fields.String(required=True)
+    active_version_id = fields.UUID(required=True)
+    versions = fields.Nested(ReportVersionSchema, required=True, many=True)
+
+    @pre_load
+    def add_report_id_to_context(self, data):
+        if "report_id" in data:
+            self.context["report_id"] = data["report_id"]
+        return data
+
+    @post_load
+    def make_versioned_report(self, data):
+        return VersionedReport(**data)
+
+
+def _relative_project_path(path):
+    if path.startswith("/project/"):
+        return path[8:]
+    else:
+        return path
+
+
+class ReportClient(BaseClient):
+
+    SERVICE_NAME = "tavern"
+
+    def list(self, project_id):
+        endpoint = "/project/{}".format(project_id)
+        return self._get(endpoint, ReportSchema(many=True))
+
+    def list_report_versions(self, report_id):
+        endpoint = "/report/{}/versions".format(report_id)
+        return self._get(endpoint, VersionedReportSchema())
+
+    def get(self, report_id):
+        endpoint = "/report/{}/active".format(report_id)
+        return self._get(endpoint, ReportSchema())
+
+    def create(
+        self,
+        name,
+        notebook_path,
+        project_id,
+        author_id,
+        description=None,
+        show_code=False,
+    ):
+        notebook_path = _relative_project_path(notebook_path)
+
+        payload = {
+            "report_name": name,
+            "author_id": author_id,
+            "notebook_path": notebook_path,
+            "description": description if description is not None else "",
```

Comment:
I looked at Tavern and I believe that passing `None` here (mapped to `null` in JSON) is accepted - I would recommend doing that (assuming it works against a real tavern).